In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [2]:
df = pd.read_csv('youtube_dislike_dataset.csv')

# 1. Import required libraries and read the provided dataset (youtube_dislike_dataset.csv) and retrieve top5 and bottom 5 records.

In [3]:
df.head(5)

video_id                                       title  \
0  --0bCF-iK2E        Jadon Sancho  Magical Skills & Goals   
1  --14w5SOEUs          Migos - Avalanche (Official Video)   
2  --40TEbZ9Is  Supporting Actress in a Comedy: 73rd Emmys   
3  --4tfbSyYDE     JO1'YOUNG (JO1 ver.)' PERFORMANCE VIDEO   
4  --DKkzWVh-E                Why Retaining Walls Collapse   

                 channel_id          channel_title         published_at  \
0  UC6UL29enLNe4mqwTfAyeNuw             Bundesliga  2021-07-01 10:00:00   
1  UCGIelM2Dj3zza3xyV3pL3WQ              MigosVEVO  2021-06-10 16:00:00   
2  UClBKH8yZRcM4AsRjDVEdjMg     Television Academy  2021-09-20 01:03:32   
3  UCsmXiDP8S40uBeJYxvyulmA                    JO1  2021-03-03 10:00:17   
4  UCMOqf8ab-42UUQIdVoKwjlQ  Practical Engineering  2021-12-07 13:00:00   

   view_count   likes  dislikes  comment_count  \
0     1048888   19515       226           1319   
1    15352638  359277      7479          18729   
2      925281   11212       401            831   
3     2641597   39131       441           3745   
4      715724   32887       367           1067   

                                                tags  \
0  football soccer ftbol alemn Bundesliga season ...   
1  Migos Avalanche Quality Control Music/Motown R...   
2                                                      
3  PRODUCE101JAPAN              JO1   TheSTAR STA...   
4  retaining wall New Jersey highway Direct Conne...   

                                         description  \
0  Enjoy the best skills and goals from Jadon San...   
1  Watch the the official video for Migos - "Aval...   
2  Hannah Waddingham wins the Emmy for Supporting...   
3  JO1'YOUNG (JO1 ver.)' PERFORMANCE VIDEO\n\n---...   
4  One of the most important (and innocuous) part...   

                                            comments  
0  Respect to Dortmund fans,must be sad losing hi...  
1  Migos just makes me want to live my live to th...  
2  Hannah's energy bursts through any screen. Wel...  
3  youngVer><  REN is really PERFECT. It's not ju...  
4   Keep up with all my projects here: https://pr...

In [4]:
df.tail(5)

video_id                                              title  \
37417  zzd4ydafGR0  Lil Tjay - Calling My Phone (feat. 6LACK) [Off...   
37418  zziBybeSAtw  PELICANS at LAKERS | FULL GAME HIGHLIGHTS | Ja...   
37419  zzk09ESX7e0                 [MV]  (MAMAMOO) - Where Are We Now   
37420  zzmQEb0Em5I              FELLIPE ESCUDERO- Master Podcast  #12   
37421  zzxPZwaA-8w  Gareth Bale brace secures dramatic comeback on...   

                     channel_id      channel_title         published_at  \
37417  UCEB4a5o_6KfjxHwNMnmj54Q           Lil Tjay  2021-02-12 05:03:49   
37418  UCWJ2lWNubArHWmf3FIHbfcQ                NBA  2021-01-16 05:39:05   
37419  UCuhAUMLzJxlP1W7mEk0_6lA            MAMAMOO  2021-06-02 09:00:10   
37420  UC8NjnNWMsRqq11NYvHAQb1g     Master Podcast  2020-10-20 20:59:30   
37421  UCEg25rdRZXg32iwai6N6l0w  Tottenham Hotspur  2021-05-23 21:00:31   

       view_count    likes  dislikes  comment_count  \
37417   120408275  2180780     35871          81360   
37418     2841917    20759      1049           2624   
37419    13346678   720854      4426          90616   
37420      252057    19198      1234           1471   
37421     2252090    34063       868           2004   

                                                    tags  \
37417  Lil Tjay Steady Calling My Phone Calling My Ph...   
37418  NBA G League Basketball game-0022000187 Lakers...   
37419  MAMAMOO  WAW  WAW MAMAMOO WAW Where Are We Now...   
37420  master masterpodcast lord lord vinheteiro z z ...   
37421  Spurs Tottenham Hotspur   Tottenham Leicester ...   

                                             description  \
37417  Official video for "Calling My Phone" by Lil T...   
37418  PELICANS at LAKERS | FULL GAME HIGHLIGHTS | Ja...   
37419  [MV]  (MAMAMOO) - Where Are We Now\n\nInstagra...   
37420  DOCTOR HAIR\nhttps://www.thedoctorhair.com/?fb...   
37421  Two minute highlights from Tottenham Hotspur's...   

                                                comments  
37417  'DESTINED 2 WIN' OUT NOW !! https://liltjay.ln...  
37418  Montrezl Harrell is going crazy with the rebou...  
37419  I honestly do not know why this song hit so ha...  
37420  Foi um prazer passar esta tarde com vocs debat...  
37421  Thank you Kane for everything you have given t...

# 2. Check the info of the dataframe and write your inferences on data types and shape of the dataset.

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 37422 entries, 0 to 37421
Data columns (total 12 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   video_id       37422 non-null  object
 1   title          37422 non-null  object
 2   channel_id     37422 non-null  object
 3   channel_title  37422 non-null  object
 4   published_at   37422 non-null  object
 5   view_count     37422 non-null  int64 
 6   likes          37422 non-null  int64 
 7   dislikes       37422 non-null  int64 
 8   comment_count  37422 non-null  int64 
 9   tags           37422 non-null  object
 10  description    37422 non-null  object
 11  comments       37264 non-null  object
dtypes: int64(4), object(8)
memory usage: 3.4+ MB


## <b>Inferences (Info of the data)<b>
*<n>Video Information: The data contains information about 37422 YouTube videos. Each video is identified by a unique"video_id".<n>
    
*<n>Channel Information: The videos are associated with different channels, and each channel is identified by a unique "channel_id". The name of the channel is also recorded in the "channel_title" column.<n>
    
*<n>Publication Date: The publication date of each video is recorded in the "published_at" column.<n>
    
*<n>View and Engagement Counts: The data contains information about the number of views, likes, dislikes, and comments for each video. All of these columns contain non-null values, indicating that there is data for all 37422 videos.<n>
    
*<n>Video Metadata: The data also contains information about the "title", "tags", and "description" of each video. There are also comments recorded in the "comments" column, but the number of non-null values is slightly lower at 37264.<n>
    
*<n>These are just some of the inferences that can be made from the data. By analyzing the information in each column, you can gain a deeper understanding of the content of the data and make more informed decisions about how to use it.<n>

In [6]:
df.dtypes

video_id         object
title            object
channel_id       object
channel_title    object
published_at     object
view_count        int64
likes             int64
dislikes          int64
comment_count     int64
tags             object
description      object
comments         object
dtype: object

## <n> Inferences (Data types)<n>
*<n>from dtype data two type of data set there are object type and integer type.<n>
    
*<n>object type of columns are ('video_id', 'title', 'channel_id', 'channel_title', 'published_at','tags',
       'description', 'comments').<n>
    
*<n>integer type of columns are ('view_count', 'likes', 'dislikes', 'comment_count').<n>

In [7]:
df.shape

(37422, 12)

# <n>Inferences (Shape of the data)<b>
*<n>Number of Rows: The first value, 37422, represents the number of rows in the data. This tells us that the data contains information about 37422 instances or observations.<n>

*<n>Number of Columns: The second value, 12, represents the number of columns in the data. This tells us that the data contains information about 12 variables or features for each of the 37422 observations.<n>

# 3. Check for the Percentage of the missing values and drop or impute them.

In [53]:
null_values = df['comments'].isnull().value_counts(normalize=1)*100
print(null_values[1]) # Null_persentage

0.4222115333226444


# 4. Check the statistical summary of both numerical and categorical columns and write your inferences.

In [55]:
print("Statistical Summary of Numerical Columns:")
df.describe().T

Statistical Summary of Numerical Columns:


count          mean           std      min       25%  \
view_count     37422.0  5.697838e+06  2.426622e+07  20368.0  512297.0   
likes          37422.0  1.668147e+05  5.375670e+05      0.0   13233.5   
dislikes       37422.0  4.989862e+03  3.070824e+04      0.0     281.0   
comment_count  37422.0  9.924930e+03  1.171003e+05      0.0     900.0   

                     50%         75%           max  
view_count     1319078.5  3670231.25  1.322797e+09  
likes            42330.5   130469.75  3.183768e+07  
dislikes           796.0     2461.75  2.397733e+06  
comment_count     2328.0     6184.00  1.607103e+07

# <n>Inferences (Statistical of Numerical Columns)<n>
<n>Data Range: The "count" value for each column tells us that all columns have the same number of non-null values, which is 37422. The "min" and "max" values give us an idea of the range of the data, with the "min" value representing the smallest observation and the "max" value representing the largest observation.<n>

<n>Mean and Standard Deviation: The "mean" and "std" values provide information about the central tendency and variability of the data. The mean is the average value of all observations, while the standard deviation measures how much the values in the data set deviate from the mean.<n>

<n>Percentiles: The "25%", "50%", and "75%" values represent the 25th, 50th, and 75th percentiles, respectively. These give us a rough idea of the distribution of the data and can be used to identify outliers and skewness in the data.<n>

<n>Comparison of Columns: By comparing the mean, standard deviation, and percentiles for each column, we can make inferences about the relative size and spread of the data for each column.<n>

In [56]:
for i in df.select_dtypes(include=['object']):
    print(f"{i}: {df[i].nunique()}")

video_id: 37422
title: 37113
channel_id: 10961
channel_title: 10883
tags: 28799
description: 35630
comments: 37264
published_month: 12


# <n>Inferences (Statistical of Categorical Columns)<b>
*<n>Unique Identifiers: The "video_id" column has the highest number of categories, which suggests that each video has a unique identifier.<n>

*<n>Missing Values: The difference between the number of categories for each column and the total number of observations (37422) indicates the number of missing values for each column. For example, the "title" column has 37113 categories, which means there are 309 missing values.<n>

*<n>High Cardinality: A high number of categories in a column relative to the total number of observations suggests a high cardinality in the data. For example, the "channel_id" column has 10961 categories, which is less than the total number of observations and suggests that there are a smaller number of channels relative to the number of videos.<n>

*<n>Temporal Data: The "published_at" column has 36772 categories, which means there are 651 missing values. The high number of categories in this column suggests that the data includes information about the publishing time of each video.<n>

*<n>Text Data: The "description" and "comments" columns have 35630 and 37264 categories, respectively. These columns likely contain text data and the difference in categories between the two columns suggests the number of missing values in each.<n>

# 5. Convert datatype of column published_at from object to pandas datetime.

In [13]:
df['published_at'] = pd.to_datetime(df['published_at'])

In [14]:
print(df['published_at'].dtype)

datetime64[ns]


In [15]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 37422 entries, 0 to 37421
Data columns (total 12 columns):
 #   Column         Non-Null Count  Dtype         
---  ------         --------------  -----         
 0   video_id       37422 non-null  object        
 1   title          37422 non-null  object        
 2   channel_id     37422 non-null  object        
 3   channel_title  37422 non-null  object        
 4   published_at   37422 non-null  datetime64[ns]
 5   view_count     37422 non-null  int64         
 6   likes          37422 non-null  int64         
 7   dislikes       37422 non-null  int64         
 8   comment_count  37422 non-null  int64         
 9   tags           37422 non-null  object        
 10  description    37422 non-null  object        
 11  comments       37264 non-null  object        
dtypes: datetime64[ns](1), int64(4), object(7)
memory usage: 3.4+ MB


# 6. Create a new column as 'published_month' using the column published_at (display the months only)


In [63]:
df['published_month'] = df['published_at'].dt.month

In [64]:
df['published_month']

0         7
1         6
2         9
3         3
4        12
         ..
37417     2
37418     1
37419     6
37420    10
37421     5
Name: published_month, Length: 37422, dtype: int64

# 7. Replace the numbers in the column published_month as names of the months i,e., 1 as 'Jan', 2 as 'Feb' and so on.....

In [18]:
month_sample = {1: 'Jan', 2: 'Feb', 3: 'Mar', 4: 'Apr', 5: 'May', 6: 'Jun',
             7: 'Jul', 8: 'Aug', 9: 'Sep', 10: 'Oct', 11: 'Nov', 12: 'Dec'}
df['published_month'] = df['published_month'].map(month_sample)

In [19]:
df['published_month']

0        Jul
1        Jun
2        Sep
3        Mar
4        Dec
        ... 
37417    Feb
37418    Jan
37419    Jun
37420    Oct
37421    May
Name: published_month, Length: 37422, dtype: object

# 8. Find the number of videos published each month and arrange the months in a decreasing order based on the video count.


In [20]:
video_counts = df['published_month'].value_counts()
video_counts

Oct    4991
Sep    4880
Nov    4851
Aug    4262
Dec    3072
Jul    2340
Jun    2316
Mar    2258
Feb    2137
Apr    2126
Jan    2108
May    2081
Name: published_month, dtype: int64

In [21]:
video_counts = video_counts.sort_values(ascending=False)
video_counts

Oct    4991
Sep    4880
Nov    4851
Aug    4262
Dec    3072
Jul    2340
Jun    2316
Mar    2258
Feb    2137
Apr    2126
Jan    2108
May    2081
Name: published_month, dtype: int64

# 9. Find the count of unique video_id, channel_id and channel_title.


In [65]:
df['video_id'].nunique()

37422

In [66]:
df['channel_id'].nunique()

10961

In [67]:
df['channel_title'].nunique()

10883

# 10. Find the top10 channel names having the highest number of videos in the dataset and the bottom10 having lowest number of videos.

In [71]:
df.head(2)

video_id                                 title  \
0  --0bCF-iK2E  Jadon Sancho  Magical Skills & Goals   
1  --14w5SOEUs    Migos - Avalanche (Official Video)   

                 channel_id channel_title        published_at  view_count  \
0  UC6UL29enLNe4mqwTfAyeNuw    Bundesliga 2021-07-01 10:00:00     1048888   
1  UCGIelM2Dj3zza3xyV3pL3WQ     MigosVEVO 2021-06-10 16:00:00    15352638   

    likes  dislikes  comment_count  \
0   19515       226           1319   
1  359277      7479          18729   

                                                tags  \
0  football soccer ftbol alemn Bundesliga season ...   
1  Migos Avalanche Quality Control Music/Motown R...   

                                         description  \
0  Enjoy the best skills and goals from Jadon San...   
1  Watch the the official video for Migos - "Aval...   

                                            comments  published_month  
0  Respect to Dortmund fans,must be sad losing hi...                7  
1  Migos just makes me want to live my live to th...                6

In [70]:
counts_of_view_video = df.groupby('channel_title')['video_id'].count()
counts_of_view_video

channel_title
   SilverName            1
   [KIMDUMB]             1
  (Inter TV channel)     1
  -    IT!               1
  - MBN                  1
                        ..
zanny                    5
zayn                     1
zefrank1                 9
zollotech               10
zoom                     1
Name: video_id, Length: 10883, dtype: int64

In [26]:
counts_of_view_video = counts_of_view_video.sort_values(ascending=False)
counts_of_view_video

channel_title
Sky Sports Football    533
The United Stand       301
BT Sport               246
NBA                    209
NFL                    162
                      ... 
Kannada Cinema           1
KanalD                   1
Kanak News               1
Kamille Ramos            1
zoom                     1
Name: video_id, Length: 10883, dtype: int64

In [27]:
top10 = counts_of_view_video.head(10)
top10

channel_title
Sky Sports Football    533
The United Stand       301
BT Sport               246
NBA                    209
NFL                    162
WWE                    122
SSSniperWolf            99
SSundee                 98
FORMULA 1               87
NHL                     86
Name: video_id, dtype: int64

In [28]:
bottom10 = counts_of_view_video.tail(10)
bottom10

channel_title
Karchez           1
Karate Combat     1
Kaptain Kuba      1
Kanye West        1
Kannur kitchen    1
Kannada Cinema    1
KanalD            1
Kanak News        1
Kamille Ramos     1
zoom              1
Name: video_id, dtype: int64

# 11. Find the title of the video which has the maximum number of likes and the title of the video having minimum likes and write your inferences.


In [29]:
max_likes = df['likes'].idxmax()
max_likes

26143

In [30]:
min_likes = df['likes'].idxmin()
min_likes

18654

In [31]:
max_title = df.loc[max_likes, 'title']
max_title

"BTS () 'Dynamite' Official MV"

In [32]:
min_title = df.loc[min_likes, 'title']
min_title

'Kim Kardashian\'s Must-See Moments on "Saturday Night Live" | E! News'

# <n>Inferences (Maximum number of likes)<b>
*<n>The maximum value in the "likes" column is associated with the row whose index value is <b>"26143"<b>.<n>

*<n>The minimum value in the "likes" column is associated with the row whose index value is <b>"18654"<b>.<n>

*<n>The string value associated with the "max_title" variable is "BTS () 'Dynamite' Official MV", which is the title of the video with the highest number of likes in the Data.<n>
    
*<n>The string value associated with the "max_title" variable is 'Kim Kardashian\'s Must-See Moments on "Saturday Night Live" | E! News', <n>which is the title of the video with the lowest number of likes in the Data.<n>


# 12. Find the title of the video which has the maximum number of dislikes and the title of the video having minimum dislikes and write your inferences.

In [33]:
max_dislikes = df['dislikes'].idxmax()
max_dislikes

13591

In [34]:
min_dislikes = df['dislikes'].idxmin()
min_dislikes

18654

In [35]:
max_title_dis = df.loc[max_dislikes, 'title']
max_title_dis

'Cuties | Official Trailer | Netflix'

In [36]:
min_title_dis = df.loc[min_dislikes, 'title']
min_title_dis

'Kim Kardashian\'s Must-See Moments on "Saturday Night Live" | E! News'

# <n>Inferences (Maximum number of dislikes)<b>
*<n>The maximum value in the "dislikes" column is associated with the row whose index value is <b>"13591".<n>

*<n>The minimum value in the "dislikes" column is associated with the row whose index value is <b>"18654".<n>

*<n>The string value associated with the "max_title" variable is "Cuties | Official Trailer | Netflix", which is the title of the video with the highest number of likes in the Data.<n>
    
*<n>The string value associated with the "max_title" variable is 'Kim Kardashian\'s Must-See Moments on "Saturday Night Live" | E! News', which is the title of the video with the lowest number of likes in the Data.<n>


# 13. Does the number of views have any effect on how many people disliked the video? Support your answer with a metric and a plot.


In [37]:
correlation = df['view_count'].corr(df['dislikes'])
correlation

0.6844687753905547

In [38]:
plt.figure( figsize=(10,5))
symbol = "*",marker=symbol
plt.scatter(df['view_count'], df['dislikes'],marker=symbol)
plt.xlabel('V i e w _ c o u n t',fontsize=20)
plt.ylabel('D i s l i k e s', fontsize=20)
plt.title('Scatter plot of Views vs. Dislikes\nPearson Correlation = {:.2f}'.format(correlation))
plt.show()

SyntaxError: cannot assign to literal (611355510.py, line 2)

# 14.Display all the information about the videos that were published in January, and mention the count of videos that were published in January.


In [39]:
january_videos = df[df['published_month'] == 'Jan']
print(january_videos)

january_video_count = january_videos.shape[0]
print("Number of videos published in January:", january_video_count)

          video_id                                              title  \
27     -2Gwm7QfBnE                                    Q&A With Naisha   
48     -4sfXSHSxzA           SURPRISING BRENT WITH HIS TIKTOK CRUSH!!   
95     -AJD1Fc5rpQ  WE ARE HAVING A BABY! | finding out i'm pregna...   
103    -AuJiwjsmWk              Do Ugly Foods Taste Worse? Taste Test   
182    -JhqO2KWr5U                                   Schlatt gets fit   
...            ...                                                ...   
37300  zmzFL5bG-jc  DEVINE MON PERSONNAGE AVANT AKINATOR ! (c'est ...   
37329  zpzjex7qwrA  Lampard Sacked Within Days Rorys Misery | Chel...   
37345  zqyv-B6mnBM                 Lil Wayne - Ain't Got Time (Audio)   
37383  zwfu1-24T7Q  PRADA Cup Day 1 | Full Race Replay | PRADA Cup...   
37418  zziBybeSAtw  PELICANS at LAKERS | FULL GAME HIGHLIGHTS | Ja...   

                     channel_id       channel_title        published_at  \
27     UCYwNMbogQFzMccPSuy-pPWg           MianTw